cmd/operator-sdk/generate/bundle/cmd.go

```diff
@@ -0,0 +1,134 @@
+}
+
+//nolint:lll
+func NewCmd() *cobra.Command {
+	c := &bundleCmd{}
+	cmd := &cobra.Command{
+		Use:   "bundle",
+		Short: "Generates bundle data for the operator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
+			}
+
+			// Generate kustomize bases, manifests, and metadata by default if no
+			// flags are set so the default behavior is "do everything".
+			fs := cmd.Flags()
+			if !fs.Changed("kustomize") && !fs.Changed("metadata") && !fs.Changed("manifests") {
+				c.kustomize = true
+				c.manifests = true
+				c.metadata = true
+			}
+
+			cfg, err := kbutil.ReadConfig()
+			if err != nil {
+				return fmt.Errorf("error reading configuration: %v", err)
+			}
+			c.setCommonDefaults(cfg)
+
+			if c.kustomize {
+				if err = c.runKustomize(cfg); err != nil {
+					log.Fatalf("Error generating bundle bases: %v", err)
+				}
+			}
+			if c.manifests {
+				if err = c.validateManifests(cfg); err != nil {
+					return fmt.Errorf("invalid command options: %v", err)
+				}
+				if err = c.runManifests(cfg); err != nil {
+					log.Fatalf("Error generating bundle manifests: %v", err)
+				}
+			}
+			if c.metadata {
+				if err = c.runMetadata(); err != nil {
+					log.Fatalf("Error generating bundle metadata: %v", err)
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().BoolVar(&c.kustomize, "kustomize", false, "Generate kustomize bases")
+	cmd.Flags().BoolVar(&c.manifests, "manifests", false, "Generate bundle manifests")
+	cmd.Flags().BoolVar(&c.metadata, "metadata", false, "Generate bundle metadata and Dockerfile")
+	cmd.Flags().BoolVar(&c.stdout, "stdout", false, "Write bundle manifest to stdout")
+
+	c.addCommonFlagsTo(cmd.Flags())
+
+	return cmd
+}
+
+// TODO(estroz): add flag to skip API metadata regeneration.
+func (c *bundleCmd) addCommonFlagsTo(fs *pflag.FlagSet) {
+	fs.StringVar(&c.operatorName, "operator-name", "", "Name of the bundle's operator")
+	fs.StringVarP(&c.version, "version", "v", "", "Semantic version of the operator in the generated bundle. "+
+		"Only set if creating a new bundle or upgrading your operator")
+	fs.StringVar(&c.inputDir, "input-dir", "", "Directory to read an existing bundle from. "+
+		"This directory is the parent of your bundle 'manifests' directory, and different from --manifest-root")
+	fs.StringVar(&c.outputDir, "output-dir", "", "Directory to write the bundle to")
+
+	fs.StringVar(&c.manifestRoot, "manifest-root", "", "Root directory for operator manifests such as "+
+		"Deployments and RBAC, ex. 'deploy' or 'config'. This directory is different from that passed to --input-dir")
+	// NB(estroz): still debating the name of this flag. For now, hide it as an
+	// "alpha" flag so we do not have to deprecate it if we change this name.
+	// TODO(estroz): decide on this flag's name before making 'init' default.
+	if err := fs.MarkHidden("manifest-root"); err != nil {
+		panic(err)
+	}
+
```

Comment:
Can we just improve the info to clarifies by giving examples over what from where is expected usually? Eg. Directory to read an existing bundle format from. Eg <default path of it in the scaffolded project>

Reply:
For `--manifest-root`? The help lists `deploy` and `config` as example values to pass. I expect we can improve this flag in #2860.